analyzers/src/SonarAnalyzer.CSharp/Rules/AvoidLambdaExpressionInLoopsInBlazor.cs

```diff
@@ -61,6 +61,6 @@ public sealed class AvoidLambdaExpressionInLoopsInBlazor : SonarDiagnosticAnalyz
                                          && semanticModel.GetSymbolInfo(invocation.Expression).Symbol is IMethodSymbol symbol
                                          && symbol.ContainingType.GetSymbolType().Is(KnownType.Microsoft_AspNetCore_Components_Rendering_RenderTreeBuilder));
 
-    private static bool IsWithinLoop(SyntaxNode node) =>
-        node.AncestorsAndSelf().Any(x => x is ForStatementSyntax or ForEachStatementSyntax or WhileStatementSyntax or DoStatementSyntax);
+    private static bool IsWithinBlockLoop(SyntaxNode node) =>
+        node.AncestorsAndSelf().Any(x => x is BlockSyntax && x.Parent is ForStatementSyntax or ForEachStatementSyntax or WhileStatementSyntax or DoStatementSyntax);
```

Comment:
The new check introduces the following False Negative:
```
    @foreach (var button in Buttons.Where(x => x.Id == "SomeId")) @* Compliant *@
        @if (button.Id == "SomeId")
        {
            <button @key="button.Id" @onclick="(e) => button.Action(e)"> @* FN *@
                Button #@button.Id
            </button>
        })
```

This is because the body of a loop statement is not necessarily a block, and can be a statement:

```cs
var i = 0;
foreach (var item in collection)
	if (true)
	{
		i++;
	}
```

![image](https://github.com/SonarSource/sonar-dotnet/assets/2676742/dc80e12f-9a51-4f84-ba2d-96d378a36aa4)

If we decide we are fine with it, we should document it in the test cases.

Reply:
I will let it as an acceptable FN.
It is not common in Razor markup to have blockless loops I would say.
And since the rule is non-SonarWay and fixing would take some time, I think we are OK to postpone the fix.
I created a ticket #8394.

I added repros in this PR.